mne_sandbox/preprocessing/_sns.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+"""Sensor noise suppression"""
+
+import numpy as np
+
+from mne import compute_raw_covariance
+from mne.io.pick import _pick_data_channels, pick_channels
+from mne.io import _BaseRaw
+from mne.utils import logger, verbose
+
+from ._dss import _pca
+
+
+class SensorNoiseSuppression(object):
+    """Apply the sensor noise suppression (SNS) algorithm
+
+    This algorithm (from [1]_) will replace the data from each channel by
+    its regression on the subspace formed by the other channels.
+
+    .. note:: Bad channels are not modified or reset by this class.
+
+    Parameters
+    ----------
+    n_neighbors : int
+        Number of neighbors (based on correlation) to include in the
+        projection.
+    reject : dict | str | None
+        Rejection parameters based on peak-to-peak amplitude.
+        See :class:`mne.Epochs` for details.
+        This is only used during the covariance-fitting phase.
+    flat : dict | None
+        Rejection parameters based on flatness of signal.
+        See :class:`mne.Epochs` for details.
+        This is only used during the covariance-fitting phase.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    References
+    ----------
+    .. [1] De Cheveigné A, Simon JZ. Sensor noise suppression. Journal of
+           Neuroscience Methods 168: 195–202, 2008.
+    """
+    @verbose
+    def __init__(self, n_neighbors, reject=None, flat=None, verbose=None):
+        self._n_neighbors = int(n_neighbors)
+        if self._n_neighbors < 1:
+            raise ValueError('n_neighbors must be positive')
```

Comment:
I usually do that, but in this case it's implied what it was (a
non-positive integer) which I figured was specific enough for users to get
how to fix it. But I'll change it for consistency